### ISSUE TITLE
Feature: Add the ability to store state

### ISSUE THREAD

**Issue by NickoOwen** [NickoOwen/bananabot#17]
Whenever BananaBot restarts the announcements are reset to the factory defaults and a new password is generated. It would be good if BananaBot was able to store state to avoid needing to reconfigure custom announcements after a restart. 

This task is to implement a mechanism to store the current state of the system (announcements and any other important variables). The preference is to store announcements in a file using a format such as JSON or perhaps YAML depending on which best suits our needs. While a database would be a viable solution, BananaBot is intended to be as lightweight as possible so the preference is to store data in a file as the state of the system should never be too massive. I would like to explore using YAML but am open to other suggestions.

This task should add the ability to store the announcements and any other required data. The storing of the password however will be part of another issue as it needs to be done following best practices. Including hashing (which already occurs) and salting the password.

**Done:**
* BananaBot now stores the announcements and other required data so it can recover and eliminate the need to reconfigure the system after reboots
* Unit tests for the new functionality

**Comment by NickoOwen** [comment]
After almost 6 months I have finally returned and finished (I think) this feature. The only remaining things to do are:
* Add validation for the config file once it is detected
* Fix the old unit tests and create new ones for the additional code

The app now checks for a config file on startup and loads the config if there is one, or creates one using the defaults if there isn't one. If the app was active when it was closed workers are created for the announcements. The config file is updated anytime a change is made to the state of the system (new announcement, banana time or text updated, days changes etc.) to ensure that in the event of a crash, the most recent state should have been saved.

**Comment by NickoOwen** [comment]
Validation was added and everything was working as intended from my testing. I began fixing the unit tests which was going well however I began to run into issues when trying to create a clean test environment for each test. Configuration seemed to be persisting across tests.

I tried a number of different solutions to fix the issues but in the end didn't have much luck. This to me highlighted a pretty major design flaw with how I was handling the app state. As I was already unhappy with the design (which I came up with ~6 months prior) it didn't take much to convince myself it needed to change.

I decided to refactor the `AppConfig` class into an `AppState` class using the Singleton Pattern. The Singleton Pattern is perfect to store the application's state as there should only ever be one, and it needs to be accessible throughout the app.

As part of this refactor, I am also making an effort to further ensure separation of responsibilities, and also fix any inconsistencies in the code (camel case vs snake case etc.)

**Comment by NickoOwen** [comment]
The work mentioned in my last comment has now been completed. Specifically:
* I have refactored `AppConfig` to `AppState` which utilises the Singleton Pattern to share state around the application
* I have refactored the logic from endpoints.py that modified the application state to utilities.py where possible
* I have fixed inconsistencies in the code, ensuring snake case is used throughout the codebase
* I have re-implemented all the test cases to work with the refactored code

All the unit tests pass and as far as I can tell in my testing of the app, everything is working as intended. Ideally, we would have a better test suite, however as this is a small personal project we are not always going to be able to have what would be ideal.

I think this task might FINALLY be done after letting it sit on a branch for 6 months. I will be making a PR shortly :)
